src/pcre2grep.c

```diff
@@ -1539,12 +1539,13 @@ switch(endlinetype)
   for (;;)
     {
     while (p < endptr && *p != '\r') p++;
-    if (++p >= endptr)
+    if (p == endptr)
```

Comment:
The code change here should absolutely no effect.

However, it silences the Clang warning. The old code would do `++p` after p had already reached endptr, which is a two-past-the-end pointer.

Clang notices that this is Undefined Behaviour (you can legally construct a one-past-the-end pointer, but no further).

My updated code here simply ensures that `p < endptr` before we increment it.